File: app/src/main/java/com/riders/thelab/data/local/dao/WeatherDao.java
package com.riders.thelab.data.local.dao;

import android.database.Cursor;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import com.riders.thelab.data.local.model.weather.CityModel;
import com.riders.thelab.data.local.model.weather.WeatherData;

import java.util.List;

import io.reactivex.Maybe;
import io.reactivex.Single;

import static androidx.room.OnConflictStrategy.REPLACE;

@Dao
public interface WeatherDao {

    @Insert(onConflict = REPLACE)
    Maybe<Long> insertRX(CityModel city);

    @Insert(onConflict = REPLACE)
    Maybe<List<Long>> insertAllRX(List<CityModel> cities);

    /* Method to insert contacts fetched from api to room */
    @Insert(onConflict = REPLACE)
    void insert(CityModel city);

    @Insert(onConflict = REPLACE)
    Maybe<Long> insert(WeatherData isWeatherData);


    /* Method to fetch contacts stored locally */
    @Query("SELECT * FROM weather_data")
    Single<WeatherData> getWeatherData();

    @Query("SELECT * FROM city")
    Single<List<CityModel>> getCities();

    @Query("SELECT * FROM city WHERE name LIKE :cityText")
    LiveData<List<CityModel>> getCities(String cityText);

    @Query("SELECT * FROM city WHERE name LIKE :cityText")
    Cursor getCitiesCursor(String cityText);


    @Query("DELETE FROM city")
    void deleteAll();
}

File: app/src/main/java/com/riders/thelab/data/local/dao/WeatherDao.kt
package com.riders.thelab.data.local.dao

import android.database.Cursor
import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.riders.thelab.data.local.model.weather.CityModel
import com.riders.thelab.data.local.model.weather.WeatherData

@Dao
interface WeatherDao {
    /* Method to insert contacts fetched from api to room */
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(city: CityModel)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertRX(city: CityModel): Long

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllRX(cities: List<CityModel>): List<Long>


    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(isWeatherData: WeatherData): Long


    /* Method to fetch contacts stored locally */
    @Query("SELECT * FROM weather_data")
    fun getWeatherData(): WeatherData

    @Query("SELECT * FROM city")
    fun getCities(): List<CityModel>

    @Query("SELECT * FROM city WHERE name LIKE :cityText")
    fun getCities(cityText: String): LiveData<List<CityModel>>

    @Query("SELECT * FROM city WHERE name LIKE :cityText")
    fun getCitiesCursor(cityText: String): Cursor


    @Query("DELETE FROM city")
    fun deleteAll()
}